beacon_node/beacon_chain/src/early_attester_cache.rs

```diff
@@ -70,14 +70,23 @@ impl<E: EthSpec> EarlyAttesterCache<E> {
             },
         };
 
-        let (_, block, blobs, data_columns) = block.deconstruct();
+        let (blobs, data_columns) = match block.data() {
+            AvailableBlockData::NoData => (None, None),
+            AvailableBlockData::Blobs(blobs) => (Some(blobs.clone()), None),
+            AvailableBlockData::DataColumns(data_columns) => (None, Some(data_columns.clone())),
+            // TODO(das): Once the columns are received, they will not be available in
+            // the early attester cache. If someone does a query to us via RPC we
+            // will get downscored.
+            AvailableBlockData::DataColumnsRecv(_) => (None, None),
```

Comment:
Good catch - this is an existing issue. I'll raise an issue tomorrow

Reply:
I think there's only little window where the downscoring can happen - once we finish computing then we'll be able to serve from the db

Reply:
This is a very small window (~200ms) I'm not sure if it's worth doing this, inclined to leave it as it is and revisit if necessary.